plugins/placeholdertext/plugin.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @fileOverview Simple CKEditor 4 plugin that adds placeholder text to the editor.
+ */
+( function() {
+	CKEDITOR.plugins.add( 'placeholdertext', {
+		isSupportedEnvironment: function() {
+			return !CKEDITOR.env.ie || CKEDITOR.env.version >= 9;
+		},
+
+		onLoad: function() {
+			CKEDITOR.addCss( CKEDITOR.plugins.placeholdertext.styles );
+		},
+
+		init: function( editor ) {
+			if ( !this.isSupportedEnvironment() || !editor.config.placeholdertext ) {
+				return;
+			}
+
+			bindPlaceholderEvents( editor, [
+				'contentDom',
+				'focus',
+				'blur',
+				'change'
```

Comment:
`change` event is triggered pretty often. I'm not totally sure how big performance impact has `editor.getData` method used by the toggling function and overall the rest of the toggling implementation, but maybe it makes sense to throttle this function a bit? We could use one of the [throttle](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_tools_buffers_throttle.html) implementations for that. WDYT?

Reply:
TBH I'd skip adding throttle unless there are real performance issues with the current solution.